include/libeconf.h

```diff
@@ -21,117 +22,302 @@
+ */
+typedef struct econf_file econf_file; /**< econf_file struct, see keyfile.h */
+
+/**
+ * @brief Generic macro calls @ref setter depending on the value type.
+ * @def econf_setValue(kf, group, key, value)
+ * @param key_file The econf_file struct.
+ * @param group The group in which the key is present.
+ * @param key The key to modify.
+ * @param value The value to set.
+ *
+ * @note Usage:
+ * @code
+ *      econf_setValue(econf_file *key_file, char *group, char *key, _generic_ value);
+ * @endcode
+ * Replace _generic_ with one of the supported value types.
+ * Supported Types: int, long, unsigned int, unsigned long, float, double,
+ *                  string (as *char).
+ *
+ * @warning Does not detect "yes", "no", 1 and 0 as boolean type. If you want to
+ *       set a bool value use "true" or "false" or use setBoolValue() directly.
+ */
+#define econf_setValue(key_file, group, key, value) ((         \
+    _Generic((value),                                          \
+    int: econf_setIntValue,                                    \
+    long: econf_setInt64Value,                                 \
+    unsigned int: econf_setUIntValue,                          \
+    unsigned long: econf_setUInt64Value,                       \
+    float: econf_setFloatValue,                                \
+    double: econf_setDoubleValue,                              \
     char*: econf_setStringValue, void*: econf_setStringValue)) \
-(kf, group, key, value))
-
-/* Generic macro to free memory allocated by econf_ functions
-   Use: econf_free(_generic_ value);
-   Replace _generic_ with one of the supported value types.
-   Supported Types: char** and econf_file*.  */
-#define econf_free(value) (( \
-  _Generic((value), \
+(key_file, group, key, value))
+
+
+
+/**
+ * @brief Generic macro to free memory allocated by econf_ functions.
+ * @note Usage:
+ * @code
+ *      econf_free(_generic_ value);
+ * @endcode
+ * Replace _generic_ with one of the supported value types.
+ * Supported Types: char** and econf_file*.
+ */
+#define econf_free(value) ((      \
+    _Generic((value),             \
     econf_file*: econf_freeFile , \
-    char**: econf_freeArray)) \
+    char**: econf_freeArray))     \
 (value))
 
-typedef struct econf_file econf_file;
 
-// Process the file of the given file_name and save its contents into key_file
+/**
+ * @brief Process the file of the given file_name and save its contents into a key_file.
+ * @param result The key file where the content is saved.
+ * @param file_name The file name of the file to process.
+ * @param delimiter The delimiter used in the file.
+ * @param comment The character used for comments.
+ * @result An econf_err error code.
+ */
 extern econf_err econf_readFile(econf_file **result, const char *file_name,
-				    const char *delim, const char *comment);
+				const char *delimiter, const char *comment);
 
-// Merge the contents of two key files
+/**
+ * @brief Merge the contents of two key files.
```

Comment:
Again a bit more flesh please, this needs more explanation. What happens if a value is set in both files - which entry wins?